internal/sirius/client.go

```diff
@@ -111,6 +112,21 @@ type ValidationError struct {
 	Field  FieldErrors `json:"validation_errors"`
 }
 
+type SiriusFieldErrors map[string][]string
+
+type SiriusValidationError struct {
```

Comment:
I'm confused by this, if it returns the errors as `validation_errors` what was the problem with using `ValidationError`?

Reply:
This will be slightly annoying (and hard to explain) so hopefully this example makes some sense, and you can adapt it to what is needed here

```go
type FlexibleFields map[string]json.RawMessage
type StrictFields map[string]map[string]string

func toStrict(x FlexibleFields) StrictFields {
	s := StrictFields{}
	for k, v := range x {
		var asMap map[string]string
		if err := json.Unmarshal(v, &asMap); err == nil {
			s[k] = asMap
			continue
		}

		var asSlice []string
		if err := json.Unmarshal(v, &asSlice); err == nil {
			s[k] = map[string]string{"": strings.Join(asSlice, "")}
			continue
		}

		panic("could not parse error") // but should just make toStrict return an error
	}
	return s
}

func main() {
	var x FlexibleFields
	json.Unmarshal([]byte(`{"field": {"reason": "desc"}}`), &x)
	var y FlexibleFields
	json.Unmarshal([]byte(`{"field": ["desc"]}`), &y)

	fmt.Println(toStrict(x), toStrict(y))
	// => map[field:map[reason:desc]] map[field:map[:desc]]
}
```

Reply:
🤦‍♀️ good point, i'll give that a go!

Reply:
Another thing (that you can ignore if you want) is that we never use the inner map key, so maybe the best representation for `FieldErrors` would be `map[string][]string` and we should transform the `map[string]map[string]string` version to that instead. That way you wouldn't have to put in the `strings.Join` "hack"

Reply:
I looked into this but it seems more effort to always need to transform the validation errors returned from `map[string]map[string]string` to `map[string][]string` considering most of the time it will be in that format 

```
	var asFieldErrors map[string]map[string]string
	if err := json.Unmarshal(v, &asFieldErrors); err == nil {
		for key, value := range asFieldErrors {
			for _, value1 := range value {
				s[key] = []string{value1}
			}
		}
		continue
	}
```

Reply:
Yeah that makes sense